File: 3952-trionic-array-i/trionic-array-i.cpp
class Solution {
public:
    bool isTrionic(vector<int>& nums) {
        int i = 1;
        if(nums[1] <= nums[0] || nums[nums.size() - 1] <= nums[nums.size() - 2]) return false;
        while(i < nums.size() && nums[i] > nums[i - 1]) i++;
        if(i == nums.size()) return false;
        while(i < nums.size() && nums[i] < nums[i - 1]) i++;
        while(i < nums.size() && nums[i] > nums[i - 1]) i++;
        if(i != nums.size()) return false;
        return true;
    }
};

File: 3952-trionic-array-i/trionic-array-i.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isTrionic = function(nums) {
    const n = nums.length;
    let i = 1;
    while (i < n && nums[i] > nums[i - 1]) i++;
    if (i === 1 || i === n) return false;
    while (i < n && nums[i] < nums[i - 1]) i++;
    if (i === n) return false;
    while (i < n && nums[i] > nums[i - 1]) i++;
    return i === n;
};
